### PR TITLE
Added initial impl for IITO 

### PR DESCRIPTION
**Pull request by NGovani**
So this is my initial implementation write-up. Along with grammar and the usual concerns please feel free to make any comments on content and whether or not you think what is written is appropriate for the implementation section. Also if there are any additions you think would be beneficial that would also be appreciated

### CODE REVIEW

**Comment by ezgimez** [comment]
wow, such a tinnnyyy PR -- let me read

File: 06_iito/index.tex [hunk 1]
```diff
@@ -1,9 +1,12 @@
 \chapter{Inter Island Trade Organisation (IITO)}
```
Comment (ezgimez) [inline 06_iito/index.tex]: let's change the title meanwhile:
"Inter-Island .."

yes, the dash is missing

File: 06_iito/index.tex [hunk 2]
```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
...
+\label{subsec:IITO:server_client_infrastructure}  
+
+All IITO sessions, such as Gift Giving or Common Pool Donations, follow the same client server interaction format. The server acts a middleman for all agent to agent communication, and prompts agents to make decisions or formulate messages during the session. At each stage in a session the server compiles a list of messages from the agents, following that sessions' protocols, and then uses that list as input for the next stage. While a bit abstract this is better explained through an example.
+
+\subsection{Gifting Session}
+\label{subsec:IITO:gifting_session} 
+
+There are four phases to the gifting sessions: Gift requests, Gift offers, Gift responses and Updating gift history. Each part is further explained below and Figure~\ref{fig:IITO:gifting_session_diagram} shows the interactions a single client will have with the server over the session.
+\begin{itemize}
+    \item \textbf{Gift Requests:} Collects the gift requests of all islands. These requests include an amount the client wants and who they wish to request from.
```
Comment (ezgimez) [inline 06_iito/index.tex]: get the semicolon out of \textbf{}.

suggestion: \textbf{Gift Requests}:

File: 06_iito/index.tex [hunk 3]
```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
...
\ No newline at end of file
+Islands are also able to offer gifts without a request being made, meaning they can reach out to other islands if they want to boost their popularity.
+
+\section{Implementation}
+\label{sec:IITO:Implementation}
+
+\subsection{Server Client Infrastructure}
+\label{subsec:IITO:server_client_infrastructure}  
+
+All IITO sessions, such as Gift Giving or Common Pool Donations, follow the same client server interaction format. The server acts a middleman for all agent to agent communication, and prompts agents to make decisions or formulate messages during the session. At each stage in a session the server compiles a list of messages from the agents, following that sessions' protocols, and then uses that list as input for the next stage. While a bit abstract this is better explained through an example.
```
Comment (ezgimez) [inline 06_iito/index.tex]: "middleman" ---> 'intermediary' ??

(I don't like the words that contain -man at the end,... but ok :D )..

also NOT capital letters for gift giving or common pool donations. (now you have them capitalised..)

File: 06_iito/index.tex [hunk 4]
```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
...
+
+\subsection{Gifting Session}
+\label{subsec:IITO:gifting_session} 
+
+There are four phases to the gifting sessions: Gift requests, Gift offers, Gift responses and Updating gift history. Each part is further explained below and Figure~\ref{fig:IITO:gifting_session_diagram} shows the interactions a single client will have with the server over the session.
```
Comment (ezgimez) [inline 06_iito/index.tex]: suggested re-writing:

There are four phases to the gifting sessions: **Gift requests, gift offers, gift responses and updating gift history**.

no unnecessary capital letters basically :)

Comment (ezgimez) [reply 06_iito/index.tex]: also is it ''session'' ??
or turn/round??

File: 06_iito/index.tex [hunk 5]
```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
...
+\begin{itemize}
+    \item \textbf{Gift Requests:} Collects the gift requests of all islands. These requests include an amount the client wants and who they wish to request from.
+    \item \textbf{Gift Offers:} Gives the clients any requests that are directed to them and then collects the gift offers from each client. These offers include an amount the client wished to gift and to whom the gift is being offered.
```
Comment (ezgimez) [inline 06_iito/index.tex]: make a full-sentence -- do NOT start with the verb directly. (looks bad)

clients ---> islands (terminology 'client' has a different meaning in infra context thoo.... but here, you mean islands)

suggested re-writing:
\item \textbf{Gift Offers}:  **It gives the islands**

File: 06_iito/index.tex [hunk 6]
```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
...
+    \item \textbf{Gift Responses:} Provides the client with all the offers directed towards them and prompts them to respond to each of these offers. These responses contain the amount the client wishes to accept and a reason for their decision.
```
Comment (ezgimez) [inline 06_iito/index.tex]: again,

clients --> islands.

**Provides .**..  --> do NOT start with the verb directly

also put the semicolon out of \textbf{}
(not gonna repeat the same pattern of errors every time :) )

File: 06_iito/index.tex [hunk 7]
```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
...
+    \item \textbf{Updating Gift History:} In this stage client's are told of the outcome of any offers they gave out. A client is told whether or not an offer has been accepted, the reason for the rejection or acceptance and also the amount the receiver wishes take. For example Team 1 may offer 100 resources to Team 2, but Team 2 may only want take 50. The implemented system allows for this behaviour.
```
Comment (ezgimez) [inline 06_iito/index.tex]: go to math mode for numbers:
`$100$` ....

client --> agent.

suggested re-writing (for very last sentence):  The implemented system allows **such** behaviour.

File: 06_iito/index.tex [hunk 8]
```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
...
+\end{itemize}
+
+\begin{figure}[!htb]
+    \centering
+    \includegraphics[width=0.6\textwidth]{06_iito/images/gifting_diagram.png}
+    \caption{Overview of Client-Server Interactions in a Gifting session}
+    \label{fig:IITO:gifting_session_diagram}
```
Comment (ezgimez) [inline 06_iito/index.tex:62]: good labelling -- loved it

**Review by ezgimez — CHANGES_REQUESTED** [review]
very minor changes requested -- overall looks LEGIT to me

File: 06_iito/index.tex (hunk 4 above)
Comment (NGovani) [reply 06_iito/index.tex]: So in each round a single gift sessions occurs. I'll add something explaining that

File: 06_iito/index.tex (hunk 8 above)
Comment (ezgimez) [reply 06_iito/index.tex:62]: put a dot "." at the end of the caption..

also, a bit unnecessary capitalisation (in my opinion) :

suggested re-writing:

 \caption{Overview of client-server interactions in a gifting session.}

File: 06_iito/index.tex [hunk 9]
```diff
@@ -42,17 +42,17 @@ \section{Implementation}
...
+During a turn, once IITO has started, a series of sessions are run and the conclusion of all these sessions indicates that IITO is also complete. In the current implementation these sessions include gift giving and common pool donations and they all follow the same client server interaction format\footnote{Server and Client are defined in Definition~\ref{def:server} and Definition~\ref{def:client} respectively}. The server acts an intermediary for all client to client communication, and prompts clients to make decisions or formulate messages during the session. At each stage in a session the server compiles a list of messages from the clients, following that sessions' protocols, and then uses that list as input for the next stage. While a bit abstract this is better explained through an example.
```
Comment (ezgimez) [inline 06_iito/index.tex]: suggested re-writing:
During a turn, once IITO has started, a series of sessions are run**,** and the conclusion of all these sessions indicates that **the** IITO **is complete**. In the current implementation**,** these sessions include gift giving and common pool donations**,** and they all follow the same **client-server** interaction.

**Review by ezgimez — APPROVED** [review]
approving since there was one more little suggestion/edit

**Pull request merged**